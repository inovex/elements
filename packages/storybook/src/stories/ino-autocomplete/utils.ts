import { KeyValue } from '@inovex.de/elements';

export const AutoCompleteStoryOptions: KeyValue[] = [
  {
    key: 'hh',
    value: 'Hamburg',
  },
  {
    key: 'ber',
    value: 'Berlin',
  },
  {
    key: 'k',
    value: 'Karlsruhe',
  },
];
