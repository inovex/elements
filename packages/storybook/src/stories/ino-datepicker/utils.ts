import '@inovex.de/elements';

export function registerInlineDatepickerHandler() {
  const group = document.querySelector('ino-radio-group') as HTMLInoRadioGroupElement;
  const datepicker = document.querySelector('.datepicker-group ino-datepicker') as HTMLInoDatepickerElement;

  const today = document.querySelector('.today');
  const thisWeek = document.querySelector('.thisWeek');
  const lastWeek = document.querySelector('.lastWeek');

  const todayClickHandler = () => {
    datepicker.value = '02.03.2020';
    datepicker.label = 'Am';
    group.value = 'at';
    datepicker.range = false;
  };

  const thisWeekClickHandler = () => {
    group.value = 'range';
    datepicker.label = 'Zeitraum';
    datepicker.range = true;
    datepicker.value = '02.03.2020 bis 08.03.2020';
  };

  const lastWeekClickHandler = () => {
    group.value = 'range';
    datepicker.label = 'Zeitraum';
    datepicker.range = true;
    datepicker.value = '24.02.2020 bis 01.03.2020';
  };

  const checkedChangeHandler = (e: CustomEvent<boolean>) => {
    const target = e.target as HTMLInoRadioElement;

    switch (target.value) {
      case 'at':
        datepicker.label = 'Am';
        datepicker.range = false;
        break;
      case 'after':
        datepicker.label = 'Ab';
        datepicker.range = false;
        if (group.value == 'range') {
          datepicker.value = datepicker.value.split(' - ').shift();
        }
        break;
      case 'before':
        datepicker.label = 'Bis';
        datepicker.range = false;
        if (group.value == 'range') {
          datepicker.value = datepicker.value.split(' - ').pop();
        }
        break;
      case 'range':
        datepicker.range = true;
        datepicker.label = 'Zeitraum';
        break;
    }
    group.value = target.value;
  };

  today.addEventListener('click', todayClickHandler);
  thisWeek.addEventListener('click', thisWeekClickHandler);
  lastWeek.addEventListener('click', lastWeekClickHandler);
  group.addEventListener('checkedChange', checkedChangeHandler);

  return () => {
    today.removeEventListener('click', todayClickHandler);
    thisWeek.removeEventListener('click', thisWeekClickHandler);
    lastWeek.removeEventListener('click', lastWeekClickHandler);
    group.removeEventListener('checkedChange', checkedChangeHandler);
  };
}
