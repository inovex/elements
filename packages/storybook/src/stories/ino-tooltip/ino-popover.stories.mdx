import { Meta, Description } from "@storybook/addon-docs/blocks";
import readme from "../../../../elements/src/components/ino-tooltip/readme.md";

<Meta title="Components (WIP)/ino-tooltip/Documentation" />

<Description markdown={readme} />
