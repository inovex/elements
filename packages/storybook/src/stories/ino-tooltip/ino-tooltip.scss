.story-tooltip {
  display: flex;
  justify-content: center;

  h4 {
    margin-top: 48px;
  }

  ino-button {
    margin: 12px 24px;
  }
}
