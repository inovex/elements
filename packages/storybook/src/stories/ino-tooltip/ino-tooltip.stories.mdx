import { Meta, Description } from "@storybook/addon-docs";
import readme from "../../../../elements/src/components/ino-tooltip/readme.md";

<Meta title="Notification/ino-tooltip/Documentation" />

<Description markdown={readme} />
