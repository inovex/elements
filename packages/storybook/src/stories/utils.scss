/**
 * Applies styling to the container of a story.
 * The styling is encapsulated within this mixin because the ID may change in the future.
 */
@mixin story-container {
  #root-inner {
    @content;
  }
}
