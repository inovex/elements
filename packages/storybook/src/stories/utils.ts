import { html } from 'lit-html';
import ICONS from '../../../elements/src/components/ino-icon/icons';

export const maybeCreateStoryArgs = (story: any) => {
  if (!story.args) {
    story.args = {};
  }
  if (!story.argTypes) {
    story.argTypes = {};
  }
};

export const withIconControl = (
  story: any,
  propertyName: string,
  defaultValue?: string
) => {
  maybeCreateStoryArgs(story);
  story.args[propertyName] = defaultValue || '';
  story.argTypes[propertyName] = {
    control: {
      type: 'select',
    },
    options: getIcons(),
  };
};

export const withColorScheme = (
  story: any,
  propertyName: string,
  defaultValue?: string
) => {
  maybeCreateStoryArgs(story);
  story.args[propertyName] = defaultValue || '';
  story.argTypes[propertyName] = {
    control: {
      type: 'select',
    },
    options: [
      'primary',
      'secondary',
      'success',
      'warning',
      'error',
      'light',
      'dark',
    ],
  };
};

export const withSortDirection = (
  story: any,
  propertyName: string,
  defaultValue?: string
) => {
  maybeCreateStoryArgs(story);
  story.args[propertyName] = defaultValue || '';
  story.argTypes[propertyName] = {
    options: ['desc', 'asc'],
    control: {
      type: 'select',
    },
  };
};

export const getIcons = () => ICONS.filter((icon) => !icon.startsWith('_'));

export const defaultDecorator = (story, className?: string) => {
  return html` <div class="${className ?? ''}">${story()}</div> `;
};
