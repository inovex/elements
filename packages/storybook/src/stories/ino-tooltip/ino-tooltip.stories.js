import { select, text } from '@storybook/addon-knobs';

import withStencilReadme from '_local-storybookcore/with-stencil-readme';

import componentReadme from '_local-elements/src/components/ino-tooltip/readme.md';
import './ino-tooltip.scss';

export default {
  title: 'Notification/<ino-tooltip>',
  decorators: [withStencilReadme(componentReadme)],
};

export const DefaultUsage = () => /*html*/ `
    <div class="story-tooltip">
      <ino-button id="tooltip-target">Tooltip</ino-button>
      <ino-tooltip
        ino-for="${text('ino-for', 'tooltip-target')}"
        ino-label="${text('ino-label', 'This is a customaziable tooltip text.')}"
        ino-placement="${select('ino-placement', ['top', 'right', 'bottom', 'left'], 'top')}"
        ino-trigger="${select(
          'ino-trigger',
          [
            'mouseenter',
            'focus',
            'click',
            'mouseenter focus',
            'mouseenter click',
            'focus click',
            'mouseenter focus click',
          ],
          'mouseenter focus'
        )}">
      </ino-tooltip>

      <h4>Placements</h4>
      <ino-tooltip ino-label="This is a tooltip left" ino-placement="left" ino-for="tooltip-positions-target"></ino-tooltip>
      <ino-tooltip ino-label="This is a tooltip right" ino-placement="right" ino-for="tooltip-positions-target"></ino-tooltip>
      <ino-tooltip ino-label="This is a tooltip top" ino-placement="top" ino-for="tooltip-positions-target"></ino-tooltip>
      <ino-tooltip ino-label="This is a tooltip bottom" ino-placement="bottom" ino-for="tooltip-positions-target"></ino-tooltip>
      <ino-button class="placement-button" id="tooltip-positions-target">Tooltip</ino-button>

      <h4>Triggers</h4>
      <ino-button id="tooltip-mouseenter">Mouseenter</ino-button>
      <ino-tooltip ino-placement="top" ino-for="tooltip-mouseenter" ino-label="This tooltip occurs on hover"></ino-tooltip>

      <ino-button id="tooltip-focus">Focus</ino-button>
      <ino-tooltip ino-placement="top" ino-for="tooltip-focus" ino-label="This tooltip occurs on focus" ino-trigger="focus"></ino-tooltip>

      <ino-button id="tooltip-click">Click</ino-button>
      <ino-tooltip ino-placement="right" ino-for="tooltip-click" ino-label="This tooltip occurs on click" ino-trigger="click"></ino-tooltip>

      <h4>Colors</h4>
      <ino-button id="primary-tooltip" ino-color-scheme="primary">Primary</ino-button>
      <ino-tooltip ino-for="primary-tooltip" ino-label="Primary color scheme" ino-color-scheme="primary" ino-placement="top"></ino-tooltip>

      <ino-button id="secondary-tooltip" ino-color-scheme="secondary">Secondary</ino-button>
      <ino-tooltip ino-for="secondary-tooltip" ino-label="Secondary color scheme" ino-color-scheme="secondary" ino-placement="top"></ino-tooltip>

      <ino-button id="light-tooltip" ino-fill="outline" ino-color-scheme="grey">Light</ino-button>
      <ino-tooltip ino-for="light-tooltip" ino-label="Light color scheme" ino-color-scheme="light" ino-placement="top"></ino-tooltip>

      <ino-button id="transparent-tooltip" ino-fill="outline" ino-color-scheme="grey">Transparent</ino-button>
      <ino-tooltip ino-for="transparent-tooltip" ino-label="Transparent color scheme" ino-color-scheme="transparent" ino-placement="top"></ino-tooltip>
    </div>
  `;

DefaultUsage.storyName = 'Default usage';
