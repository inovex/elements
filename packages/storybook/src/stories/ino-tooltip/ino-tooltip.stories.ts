import { html } from 'lit-html';
import { defaultDecorator, withColorScheme } from '../utils';
import { Story } from '@storybook/web-components';
import { Components } from '@inovex.de/elements';
import './ino-tooltip.scss';

export default {
  title: `Notification/ino-tooltip`,
  component: 'ino-tooltip',
  decorators: [(story) => defaultDecorator(story, 'story-tooltip')],
};

export const Playground: Story<Components.InoTooltip> = (args) => html`
  <ino-button id="tooltip-target">Tooltip</ino-button>
  <ino-tooltip
    for="${args.for}"
    label="${args.label}"
    placement="${args.placement}"
    trigger="${args.trigger}"
    color-scheme="${args.colorScheme}"
  >
  </ino-tooltip>
`;
Playground.args = {
  for: 'tooltip-target',
  label: 'This is a tooltip text.',
  placement: 'bottom',
  trigger: 'mouseenter focus',
};
Playground.argTypes = {
  placement: {
    control: {
      type: 'select',
    },
    options: ['top', 'right', 'bottom', 'left'],
  },
  trigger: {
    control: {
      type: 'select',
    },
    options: [
      'mouseenter',
      'focus',
      'click',
      'mouseenter focus',
      'mouseenter click',
      'focus click',
      'mouseenter focus click',
    ],
  },
};
withColorScheme(Playground, 'colorScheme', 'transparent');

export const Placements = () => html`
  <ino-tooltip
    label="This is a tooltip left"
    placement="left"
    for="tooltip-positions-target"
  ></ino-tooltip>
  <ino-tooltip
    label="This is a tooltip right"
    placement="right"
    for="tooltip-positions-target"
  ></ino-tooltip>
  <ino-tooltip
    label="This is a tooltip top"
    placement="top"
    for="tooltip-positions-target"
  ></ino-tooltip>
  <ino-tooltip
    label="This is a tooltip bottom"
    placement="bottom"
    for="tooltip-positions-target"
  ></ino-tooltip>
  <ino-button class="placement-button" id="tooltip-positions-target"
    >Tooltip</ino-button
  >
`;

export const Triggers = () => html`
  <ino-button id="tooltip-mouseenter">Mouseenter</ino-button>
  <ino-tooltip
    placement="top"
    for="tooltip-mouseenter"
    label="This tooltip occurs on hover"
  ></ino-tooltip>

  <ino-button id="tooltip-focus">Focus</ino-button>
  <ino-tooltip
    placement="top"
    for="tooltip-focus"
    label="This tooltip occurs on focus"
    trigger="focus"
  ></ino-tooltip>

  <ino-button id="tooltip-click">Click</ino-button>
  <ino-tooltip
    placement="right"
    for="tooltip-click"
    label="This tooltip occurs on click"
    trigger="click"
  ></ino-tooltip>
`;

export const Colors = () => html`
  <ino-button id="primary-tooltip" color-scheme="primary">Primary</ino-button>
  <ino-tooltip
    for="primary-tooltip"
    label="Primary color scheme"
    color-scheme="primary"
    placement="top"
  ></ino-tooltip>

  <ino-button id="secondary-tooltip" color-scheme="secondary"
    >Secondary</ino-button
  >
  <ino-tooltip
    for="secondary-tooltip"
    label="Secondary color scheme"
    color-scheme="secondary"
    placement="top"
  ></ino-tooltip>

  <ino-button id="light-tooltip" color-scheme="grey">Light</ino-button>
  <ino-tooltip
    for="light-tooltip"
    label="Light color scheme"
    color-scheme="light"
    placement="top"
  ></ino-tooltip>

  <ino-button id="transparent-tooltip" color-scheme="grey"
    >Transparent</ino-button
  >
  <ino-tooltip
    for="transparent-tooltip"
    label="Transparent color scheme"
    color-scheme="transparent"
    placement="top"
  ></ino-tooltip>
`;
