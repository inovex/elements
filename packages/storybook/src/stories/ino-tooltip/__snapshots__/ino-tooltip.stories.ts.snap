// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Notification/ino-tooltip ColorScheme smoke-test 1`] = `
<div id="root-inner">
  <div class="story-tooltip">
    <ino-button id="tooltip-target-color-scheme">
      Trigger Element
    </ino-button>
    <ino-tooltip for="tooltip-target-color-scheme"
                 placement="auto"
                 trigger="mouseenter focus"
                 color-scheme="secondary"
    >
      This is a tooltip
    </ino-tooltip>
  </div>
</div>
`;

exports[`Notification/ino-tooltip Placement smoke-test 1`] = `
<div id="root-inner">
  <div class="story-tooltip">
    <ino-button id="tooltip-target-placement">
      Trigger Element
    </ino-button>
    <ino-tooltip for="tooltip-target-placement"
                 placement="right"
                 trigger="mouseenter focus"
                 color-scheme="primary"
    >
      This is a tooltip
    </ino-tooltip>
  </div>
</div>
`;

exports[`Notification/ino-tooltip Playground smoke-test 1`] = `
<div id="root-inner">
  <div class="story-tooltip">
    <ino-button id="tooltip-target">
      Trigger Element
    </ino-button>
    <ino-tooltip for="tooltip-target"
                 placement="auto"
                 trigger="mouseenter focus"
                 color-scheme="primary"
    >
      This is a tooltip
    </ino-tooltip>
  </div>
</div>
`;

exports[`Notification/ino-tooltip Trigger smoke-test 1`] = `
<div id="root-inner">
  <div class="story-tooltip">
    <ino-button id="tooltip-target-trigger">
      Trigger Element
    </ino-button>
    <ino-tooltip for="tooltip-target-trigger"
                 placement="auto"
                 trigger="click"
                 color-scheme="primary"
    >
      This is a tooltip
    </ino-tooltip>
  </div>
</div>
`;
